dependencies.yaml

```diff
@@ -568,24 +592,31 @@ dependencies:
         matrices:
           - matrix: {cuda: "12.*"}
             packages:
-              - rmm-cu12==24.8.*
+              - rmm-cu12==24.8.*,>=0.0.0a0
               - pynvjitlink-cu12
           - matrix: {cuda: "11.*"}
             packages:
-              - rmm-cu11==24.8.*
+              - rmm-cu11==24.8.*,>=0.0.0a0
               - cubinlinker-cu11
               - ptxcompiler-cu11
-          - {matrix: null, packages: [cubinlinker, ptxcompiler, *rmm_conda]}
+          - matrix:
+            packages:
+              - cubinlinker
+              - ptxcompiler
```

Comment:
How do we decide on whether the fallback should be CUDA 11 or 12? When can this be changed to CUDA 12 by default? If this fallback is meant to support DLFW, those builds already use CUDA 12.

Reply:
Note that this fallback was *already* `[cubinlinker, ptxcompiler, rmm==24.8.*]`... this PR is not changing the actual content here, other than adding an alpha spec for `rmm`.

It's only showing up in the diff here because the addition of the alpha spec introduced a `>`, and the YAML parser used by `rapids-dependency-file-generator` doesn't like `,>` inline in a `[]`.

Try adding a matrix entry like this:

```yaml
- {matrix: null, packages: [rmm==24.8.*,>=0.0.0a0]}
```

And running `rapids-dependency-file-generator`.

You'll see an error like this:

```text
yaml.scanner.ScannerError: while scanning for the next token
found character '>' that cannot start any token
  in "dependencies.yaml", line 328, column 51
```

---

> *When can this be changed to CUDA 12 by default? If this fallback is meant to support DLFW, those builds already use CUDA 12.*

I *think* the fallback could safely be changed to CUDA 12 dependencies at any time.

I just checked and it looks like DLFW builds are already patching `cubinlinker` and `ptxcompiler` out of `cudf`'s `pyproject.toml`.

Like, literally applying `git` patches like this:

```diff
diff --git a/python/cudf/pyproject.toml b/python/cudf/pyproject.toml
index 3112db2a72..0ebf2ca155 100644
--- a/python/cudf/pyproject.toml
+++ b/python/cudf/pyproject.toml
@@ -25,7 +25,6 @@ license = { text = "Apache 2.0" }
 requires-python = ">=3.9"
 dependencies = [
     "cachetools",
-    "cubinlinker",
     "cuda-python>=11.7.1,<12.0a0",
     "cupy-cuda11x>=12.0.0",
     "fsspec>=0.6.0",
@@ -35,7 +34,6 @@ dependencies = [
     "packaging",
     "pandas>=2.0,<2.2.2dev0",
     "protobuf>=3.20,<5",
-    "ptxcompiler",
     "pyarrow>=14.0.1,<15.0.0a0",
     "rich",
     "rmm==24.4.*",
-- 
2.43.2
```

Reply:
Thanks -- two comments.

1. We can always wrap a package specifier in quotes so that YAML interprets it as a string. I'm totally fine with the current state, though, so no change is needed.
2. Let's open a `build-planning` issue to update our "default" values from CUDA 11 to CUDA 12.

Reply:
> *totally fine with the current state, though, so no change is needed*

This entire part of the diff ended up being unnecessary anyway, if we're ok with alpha specs making it into conda env constraints: https://github.com/rapidsai/cudf/pull/15245#discussion_r1619017714.

It's no longer in the diff for this PR.

> *Let's open a build-planning issue to update our "default" values from CUDA 11 to CUDA 12.*

Created https://github.com/rapidsai/build-planning/issues/68